### ISSUE TITLE
NullPointerException when items are reloaded

### ISSUE THREAD

**Issue by boc-tothefuture** [QuailAutomation/openhab2-addons#45]
Not sure what exactly is going on, but sometimes when the items file is reloaded NPEs start on updates:
```
2017-06-25 08:55:50.988 [DEBUG] [g.omnilink.handler.ThermostatHandler] - Thermostat Status ThermostatStatus ( number = 2    status = 4    temperature = 129    heatSetpotint = 118    coolSetpoint = 130    systemMode = 3    fanMode = 0    holdStatus = 0    humidity = 93    humiditySetpoint = 72    dehumiditySetpoint = 106    outdoorTemp = 126    extendedStatus = 0     )
2017-06-25 08:55:58.142 [INFO ] [org.quartz.core.QuartzScheduler     ] - Scheduler openHAB-job-scheduler_$_NON_CLUSTERED started.
2017-06-25 08:56:04.533 [DEBUG] [nilink.handler.OmnilinkBridgeHandler] - Received Object Status Notification that was not processed: ObjectStatus ( statusType = 8    statuses = [Lcom.digitaldan.jomnilinkII.MessageTypes.statuses.AuxSensorStatus;@12d2143     )
2017-06-25 08:56:07.973 [ERROR] [om.digitaldan.jomnilinkII.Connection] - Notifcation Handler Caught Exception
java.lang.NullPointerException
        at org.openhab.binding.omnilink.handler.OmnilinkBridgeHandler.objectStausNotification(OmnilinkBridgeHandler.java:310)[248:org.openhab.binding.omnilink:2.1.0.201706251241]
        at com.digitaldan.jomnilinkII.Connection$NotificationHandler.run(Connection.java:842)[248:org.openhab.binding.omnilink:2.1.0.201706251241]
        at java.lang.Thread.run(Thread.java:748)[:1.8.0_131]
2017-06-25 08:56:11.581 [ERROR] [om.digitaldan.jomnilinkII.Connection] - Notifcation Handler Caught Exception
java.lang.NullPointerException
        at org.openhab.binding.omnilink.handler.OmnilinkBridgeHandler.objectStausNotification(OmnilinkBridgeHandler.java:310)[248:org.openhab.binding.omnilink:2.1.0.201706251241]
        at com.digitaldan.jomnilinkII.Connection$NotificationHandler.run(Connection.java:842)[248:org.openhab.binding.omnilink:2.1.0.201706251241]
        at java.lang.Thread.run(Thread.java:748)[:1.8.0_131]
2017-06-25 08:56:34.525 [DEBUG] [nilink.handler.OmnilinkBridgeHandler] - Received Object Status Notification that was not processed: ObjectStatus ( statusType = 8    statuses = [Lcom.digitaldan.jomnilinkII.MessageTypes.statuses.AuxSensorStatus;@4f55e774     )
```
Looking at the code:
```
if (theThing != null) {
                    ((ThermostatHandler) theThing.getHandler()).handleThermostatStatus(thermostatStatus);
                }
```
getHandler() is returning null.  Not sure the circumstances that would cause that. 
But I did noticed that chandleHandlerDisposed never removes our items from the mapping.  Which might leave them in an odd state on next update if we still find them in the map?

Opening this while I investigate some more.

**Comment by boc-tothefuture** [comment]
@craigham and @digitaldan - 
  Are you seeing this as well?  I essentially get into a state where all updates stop and I get nothing but NPEs:
```
java.lang.NullPointerException
        at org.openhab.binding.omnilink.handler.OmnilinkBridgeHandler.handleUnitStatus(OmnilinkBridgeHandler.java:273)[275:org.openhab.binding.omnilink:2.2.0.201707062038]
        at org.openhab.binding.omnilink.handler.OmnilinkBridgeHandler.objectStatusNotification(OmnilinkBridgeHandler.java:282)[275:org.openhab.binding.omnilink:2.2.0.201707062038]
        at com.digitaldan.jomnilinkII.Connection$NotificationHandler.run(Connection.java:842)[275:org.openhab.binding.omnilink:2.2.0.201707062038]
        at java.lang.Thread.run(Thread.java:748)[:1.8.0_131]
2017-07-06 20:52:04.275 [ERROR] [om.digitaldan.jomnilinkII.Connection] - Notification Handler Caught Exception
java.lang.NullPointerException
        at org.openhab.binding.omnilink.handler.OmnilinkBridgeHandler.objectStatusNotification(OmnilinkBridgeHandler.java:305)[275:org.openhab.binding.omnilink:2.2.0.201707062038]
        at com.digitaldan.jomnilinkII.Connection$NotificationHandler.run(Connection.java:842)[275:org.openhab.binding.omnilink:2.2.0.201707062038]
        at java.lang.Thread.run(Thread.java:748)[:1.8.0_131]
2017-07-06 20:52:07.723 [ERROR] [om.digitaldan.jomnilinkII.Connection] - Notification Handler Caught Exception
java.lang.NullPointerException
        at org.openhab.binding.omnilink.handler.OmnilinkBridgeHandler.objectStatusNotification(OmnilinkBridgeHandler.java:305)[275:org.openhab.binding.omnilink:2.2.0.201707062038]
        at com.digitaldan.jomnilinkII.Connection$NotificationHandler.run(Connection.java:842)[275:org.openhab.binding.omnilink:2.2.0.201707062038]
        at java.lang.Thread.run(Thread.java:748)[:1.8.0_131]
2017-07-06 20:52:23.326 [ERROR] [om.digitaldan.jomnilinkII.Connection] - Notification Handler Caught Exception
java.lang.NullPointerException
        at org.openhab.binding.omnilink.handler.OmnilinkBridgeHandler.objectStatusNotification(OmnilinkBridgeHandler.java:305)[275:org.openhab.binding.omnilink:2.2.0.201707062038]
        at com.digitaldan.jomnilinkII.Connection$NotificationHandler.run(Connection.java:842)[275:org.openhab.binding.omnilink:2.2.0.201707062038]
        at java.lang.Thread.run(Thread.java:748)[:1.8.0_131]
2017-07-06 20:52:27.019 [DEBUG] [ding.omnilink.handler.UpbUnitHandler] - handleCommand called
2017-07-06 20:52:27.154 [DEBUG] [nilink.handler.OmnilinkBridgeHandler] - received status update for unit: 83, status: 0
2017-07-06 20:52:27.155 [ERROR] [om.digitaldan.jomnilinkII.Connection] - Notification Handler Caught Exception
java.lang.NullPointerException
        at org.openhab.binding.omnilink.handler.OmnilinkBridgeHandler.handleUnitStatus(OmnilinkBridgeHandler.java:273)[275:org.openhab.binding.omnilink:2.2.0.201707062038]
        at org.openhab.binding.omnilink.handler.OmnilinkBridgeHandler.objectStatusNotification(OmnilinkBridgeHandler.java:282)[275:org.openhab.binding.omnilink:2.2.0.201707062038]
        at com.digitaldan.jomnilinkII.Connection$NotificationHandler.run(Connection.java:842)[275:org.openhab.binding.omnilink:2.2.0.201707062038]
        at java.lang.Thread.run(Thread.java:748)[:1.8.0_131]
2017-07-06 20:52:27.770 [DEBUG] [ding.omnilink.handler.UpbUnitHandler] - handleCommand called
2017-07-06 20:52:27.863 [DEBUG] [nilink.handler.OmnilinkBridgeHandler] - received status update for unit: 84, status: 0
2017-07-06 20:52:27.863 [ERROR] [om.digitaldan.jomnilinkII.Connection] - Notification Handler Caught Exception
java.lang.NullPointerException
        at org.openhab.binding.omnilink.handler.OmnilinkBridgeHandler.handleUnitStatus(OmnilinkBridgeHandler.java:273)[275:org.openhab.binding.omnilink:2.2.0.201707062038]
        at org.openhab.binding.omnilink.handler.OmnilinkBridgeHandler.objectStatusNotification(OmnilinkBridgeHandler.java:282)[275:org.openhab.binding.omnilink:2.2.0.201707062038]
        at com.digitaldan.jomnilinkII.Connection$NotificationHandler.run(Connection.java:842)[275:org.openhab.binding.omnilink:2.2.0.201707062038]
        at java.lang.Thread.run(Thread.java:748)[:1.8.0_131]
2017-07-06 20:53:09.439 [ERROR] [om.digitaldan.jomnilinkII.Connection] - Notification Handler Caught Exception
java.lang.NullPointerException
        at org.openhab.binding.omnilink.handler.OmnilinkBridgeHandler.objectStatusNotification(OmnilinkBridgeHandler.java:305)[275:org.openhab.binding.omnilink:2.2.0.201707062038]
        at com.digitaldan.jomnilinkII.Connection$NotificationHandler.run(Connection.java:842)[275:org.openhab.binding.omnilink:2.2.0.201707062038]
        at java.lang.Thread.run(Thread.java:748)[:1.8.0_131]
2017-07-06 20:53:16.271 [ERROR] [om.digitaldan.jomnilinkII.Connection] - Notification Handler Caught Exception
java.lang.NullPointerException
        at org.openhab.binding.omnilink.handler.OmnilinkBridgeHandler.objectStatusNotification(OmnilinkBridgeHandler.java:305)[275:org.openhab.binding.omnilink:2.2.0.201707062038]
        at com.digitaldan.jomnilinkII.Connection$NotificationHandler.run(Connection.java:842)[275:org.openhab.binding.omnilink:2.2.0.201707062038]
        at java.lang.Thread.run(Thread.java:748)[:1.8.0_131]
2017-07-06 20:53:28.467 [ERROR] [om.digitaldan.jomnilinkII.Connection] - Notification Handler Caught Exception
java.lang.NullPointerException
        at org.openhab.binding.omnilink.handler.OmnilinkBridgeHandler.objectStatusNotification(OmnilinkBridgeHandler.java:305)[275:org.openhab.binding.omnilink:2.2.0.201707062038]
        at com.digitaldan.jomnilinkII.Connection$NotificationHandler.run(Connection.java:842)[275:org.openhab.binding.omnilink:2.2.0.201707062038]
        at java.lang.Thread.run(Thread.java:748)[:1.8.0_131]
```
A restart resolves it.

**Comment by digitaldan** [comment]
I have not, but it would seem the handler for the thing is null, not sure why that would be, is there any other logs about the binding unloading or restarting?

**Comment by digitaldan** [comment]
Sorry, I did not read your original post, but I think thats exactly it, we have old things in the maps.

I actually did not notice we were caching the Things in maps,  I don't think this is necessary,  we can look up child things directly on the bridge with `getThingByUID` and not do any other caching

**Comment by boc-tothefuture** [comment]
Closing.  I believe this is resolved by #69 and #70
